languages/javalang/builder-renderer/generated-code/src/test/java/com/commercetools/ModelSerializationTest.java

```diff
@@ -0,0 +1,164 @@
+                .assets(Arrays.asList(assetDraft))
+                .custom(CustomFieldsDraftBuilder.of().type("string type").fields(fieldContainer).build())
+                .description(localizedString)
+                .externalId(id)
+                .key(key)
+                .metaDescription(localizedString)
+                .metaKeywords(localizedString)
+                .metaTitle(localizedString)
+                .name(localizedString)
+                .orderHint(randomString)
+                .slug(localizedString)
+                .build();
+        
+        String categoryDraftJson = categoryDraft.toJson();
+        
+        try{
+            final URL url = Thread.currentThread().getContextClassLoader().getResource("json_examples/category-draft-example.json");
+            String categoryDraftExample = new String(Files.readAllBytes(Paths.get(url.getPath())));
+            Assert.assertEquals(categoryDraftJson, categoryDraftExample);
+        }catch (IOException e){
+            e.printStackTrace();
+            Assert.fail();
+        }
+    }
+    
+    @Test
+    public void serializeProductDraftToJson() {
+        
+    }
+    
+    @Test
+    public void deserializeCategoryFromJson() {
+        LocalizedString localizedString = new LocalizedStringImpl();
+        localizedString.setValue("test-key", "test-value");
+        String key = "test-key";
+        String id = "test-id";
+        String randomString = "test-string";
+
+        CategoryResourceIdentifier resourceIdentifier = new CategoryResourceIdentifierImpl();
+        resourceIdentifier.setId(id);
+        resourceIdentifier.setKey(key);
+
+        FieldContainer fieldContainer = new FieldContainerImpl();
+        fieldContainer.setValue(key, randomString);
+
+        AssetDimensions assetDimensions = new AssetDimensionsImpl();
+        assetDimensions.setH(10);
+        assetDimensions.setW(5);
+
+        AssetSource assetSource = new AssetSourceImpl();
+        assetSource.setContentType("application/json");
+        assetSource.setDimensions(assetDimensions);
+
+        CustomFields customFields = new CustomFieldsImpl();
+        customFields.setFields(fieldContainer);
+        TypeReference typeReference = TypeReference.of();
+        typeReference.setKey(key);
+        typeReference.setObj(Type.of());
+        customFields.setType(typeReference);
+
+        Asset asset = new AssetImpl();
+        asset.setCustom(customFields);
+        asset.setDescription(localizedString);
+        asset.setId(id);
+        asset.setKey(key);
+        asset.setName(localizedString);
+        asset.setCustom(customFields);
+        asset.setSources(Arrays.asList(assetSource));
+        asset.setTags(Arrays.asList("tag 1", "tag 2"));
+
+        CategoryReference reference = new CategoryReferenceImpl();
+        reference.setKey(key);
+        reference.setId(id);
+        reference.setObj(new CategoryImpl());
+
+        Category category = new CategoryImpl();
```

Comment:
We didn't you use a builder here?

Reply:
Category is not a draft class so it doesn't have a builder.